src/main/scala/com/fulcrumgenomics/bam/CollectErccMetrics.scala

```diff
@@ -0,0 +1,253 @@
+) extends FgBioTool with LazyLogging {
+
+  Io.assertReadable(input)
+  Io.assertReadable(metadata)
+  Io.assertCanWriteFile(output)
+
+  private val ScriptPath = "com/fulcrumgenomics/bam/CollectErccMetrics.R"
+
+  override def execute(): Unit = {
+    val in       = SamSource(input)
+    val progress = ProgressLogger(logger)
+    val erccData = DelimitedDataParser(metadata, '\t').map { row => row[String](0) -> row[Double](1) }.toMap
+
+    // check for no ERCC data
+    if (erccData.isEmpty) {
+      throw new IllegalArgumentException(s"No ERCC data given in '$metadata'")
+    }
+
+    // require that the SAM/BAM header contains all the ERCC transcripts
+    erccData.keys.foreach { erccId =>
+      if (in.header.getSequence(erccId) == null) {
+        throw new IllegalArgumentException(s"ERCC with id '$erccId' not found in the SAM/BAM header")
+      }
+    }
+
+    var totalReads: Long    = 0 // the total # of reads in the BAM file
+    var erccReads: Long     = 0 // the total # of reads that are counted as mapped to an ERCC transcript
+    var erccFragments: Long = 0 // the total # of fragments or read pairs that are counted as mapped to an ERCC transcript
+    val erccFragmentCounter = new SimpleCounter[String]() // the count of fragments mapped to a given ERCC transcript
+    in.iterator
+      .filterNot { r => progress.record(r); r.secondary || r.supplementary } // no secondary or supplementary
+      .filter { r =>
+        totalReads += 1
+        // consider only records mapped to an ERCC transcript
+        r.mapped && r.mapq >= minimumMappingQuality && erccData.contains(r.refName)
+      }
+      .filter { r =>
+        erccReads += 1
+        // consider only paired records where both ends map to the sam ERCC transcript
+        r.paired && r.mapped && r.mateMapped && r.refIndex == r.mateRefIndex && r.firstOfPair
+      }
+      .foreach { r =>
+        erccFragments += 1
+        erccFragmentCounter.count(r.refName)
+      }
+    in.safelyClose()
+
+    // Transform to log2 counts to compute some simple correlation and linear regression metrics.
+    val counts         = ListBuffer[Double]()
+    val concentrations = ListBuffer[Double]()
+    erccData.iterator.foreach { case (name, concentration) =>
+      val count = erccFragmentCounter.countOf(name)
+      if (minTranscriptCount <= count) {
+        counts += log2(count)
+        concentrations += log2(concentration)
+      }
+    }
+
+    // Rocket science
+    val simpleRegression = new SimpleRegression()
+    counts.zip(concentrations).foreach { case (cnt, conc) => simpleRegression.addData(cnt, conc) }
+
+    val erccMetrics = ErccSummaryMetrics(
+      total_reads                = totalReads,
+      ercc_reads                 = erccReads,
+      pct_ercc_reads             = if (totalReads == 0) 0 else 100.0 * erccReads / totalReads.toDouble,
+      ercc_fragments             = erccFragments,
+      total_transcripts          = erccFragmentCounter.iterator.length,
```

Comment:
Maybe we should add a `SimpleCounter.size` that reports the number of counted things?